### PR TITLE
Add performance section regarding a span processor to the readme

### PR DESCRIPTION
**Pull request by amit-mazor**
Adding a new section to the readme about the span processor - SimpleSpanProcessor vs BatchSpanProcessor and its impact on the collector performance

### CODE REVIEW

File: otel-agent/README.md [hunk 1]
```diff
@@ -104,6 +104,22 @@ env:
     value: "$(NODE):4417"
 ```
 
+# Performance of the Collector - Picking the right span processor
+
+Picking the right span processor can have an impact on the performance of the collector.
+When using the SimpleSpanProcessor, it affects the resources usage, the latenct, and the buffering behaviour of the collector.
```
Comment (oded-dd) [inline otel-agent/README.md:110]: latenct --> latency

File: otel-agent/README.md [hunk 2]
```diff
@@ -104,6 +104,22 @@ env:
...
+
+While the SimpleSpanProcessor processes spans as they are created, the BatchSpanProcessor processes spans in batches before they are exported.
```
Comment (oded-dd) [inline otel-agent/README.md:112]: processes --> process

Comment (oded-dd) [reply otel-agent/README.md:112]: processes --> process

**Review by oded-dd — CHANGES_REQUESTED** [review]

**Comment by amit-mazor** [comment]
@oded-dd I updated the content a little bit so please re-review

**Review by oded-dd — APPROVED** [review]

**Pull request merged**